pkg/infra/tracing/tracing.go

```diff
@@ -339,7 +339,8 @@ func (ots *TracingService) Run(ctx context.Context) error {
 	if err := ots.StartAsync(ctx); err != nil {
 		return err
 	}
-	return ots.AwaitTerminated(ctx)
+	stopCtx := context.Background()
+	return ots.AwaitTerminated(stopCtx)
```

Comment:
I tested removing this just to understand its effect, but it seems regardless the service will always stop "cleanly" (according to the logs at least) whether the root ctx is canceled, if a dependency fails or when exiting with ctrl+c. I could be missing another case though - LMK! 

I see that it's tested for provisioning - was that where you questioned the idea of whether `Run` should return errors or not?